src/main/scala/nl/knaw/dans/easy/fedoratobag/EasyFedoraToBagApp.scala

```diff
@@ -219,55 +219,45 @@ class EasyFedoraToBagApp(configuration: Configuration) extends DebugEnhancedLogg
         .getOrElse(Success(()))
       _ <- managedMetadataStream(foXml, "DATASET_LICENSE", bag, "depositor-info/depositor-agreement")
         .getOrElse(Success(()))
+      isOriginalVersioned = options.transformationType == ORIGINAL_VERSIONED
       allFileInfos <- fedoraIDs.filter(_.startsWith("easy-file:")).toList.traverse(getFileInfo)
-      firstFileInfos <- selectFileInfos(options.firstFileFilter(emdXml), allFileInfos)
-      firstBagFileItems <- firstFileInfos.traverse(addPayloadFileTo(bag))
-      _ <- checkNotImplemented(firstBagFileItems, logger)
-      _ <- addXmlMetadataTo(bag, "files.xml")(filesXml(firstBagFileItems))
+      nextFileInfos = forSecondBag(allFileInfos, isOriginalVersioned)
```

Comment:
I find the naming of the variables and method calls somewhat unclear. My suggestion:

firstFileFilter -> getFilterTypeFirstBag
firstFileInfos -> fileInfosFirstBag
selectFileInfos -> getFileInfosFirstBag
firstBagFileItems -> fileItemsFirstBag
nextFileInfos -> fileInfosSecondBag
forSecondBag -> getFileInfosSecondBag